test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj

```diff
@@ -68,28 +69,58 @@
     {:status 200
      :body   "This should have timed out."}))
 
+(defprotocol TkProxyService)
+
+(defn proxy-service
+  [proxy-config proxy-opts proxy-path]
+  (service TkProxyService
+    [[:WebserverService add-proxy-route]]
+    (init [this context]
+      (if proxy-opts
+        (add-proxy-route proxy-config
+                         proxy-path
+                         proxy-opts)
+        (add-proxy-route proxy-config
+                         proxy-path))
+      context)))
+
 (defmacro with-target-and-proxy-servers
-  [{:keys [target proxy proxy-config proxy-opts ring-handler]} & body]
-  `(with-app-with-config proxy-target-app#
-     [jetty9-service]
-     {:webserver ~target}
-     (let [target-webserver# (get-service proxy-target-app# :WebserverService)]
-       (add-ring-handler
-         target-webserver#
-         ~ring-handler
-         "/hello")
-       (add-ring-handler
-         target-webserver#
-         ~ring-handler
-         "/goodbye"))
-     (with-app-with-config proxy-app#
-       [jetty9-service]
-       {:webserver ~proxy}
-       (let [proxy-webserver# (get-service proxy-app# :WebserverService)]
-         (if ~proxy-opts
-           (add-proxy-route proxy-webserver# ~proxy-config "/hello-proxy" ~proxy-opts)
-           (add-proxy-route proxy-webserver# ~proxy-config "/hello-proxy")))
-       ~@body)))
+  [{:keys [target proxy proxy-config proxy-opts ring-handler
+           register-proxy-route-before-server-start?]} & body]
+  (let [proxy-path "/hello-proxy"]
+    `(with-app-with-config proxy-target-app#
+      [jetty9-service]
+      {:webserver ~target}
+      (let [target-webserver# (get-service proxy-target-app# :WebserverService)]
+        (add-ring-handler
+          target-webserver#
+          ~ring-handler
+          "/hello")
+        (add-ring-handler
+          target-webserver#
+          ~ring-handler
+          "/goodbye"))
+      (if ~register-proxy-route-before-server-start?
+        (let [proxy-service# (proxy-service ~proxy-config
+                                            ~proxy-opts
+                                            ~proxy-path)]
+          (with-app-with-config proxy-app#
+            [jetty9-service proxy-service#]
+            {:webserver ~proxy}
+            ~@body))
+          (with-app-with-config proxy-app#
```

Comment:
This is super nitpicky so feel free to ignore it, but I was initially confused by this indentation and thought this was part of the `let` block above.